### PR TITLE
fix: unit query separated from listing query

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview
On staging environment (the hosted one) hitting certain listing pages was causing the backend to run out of memory and crash. This is a fix to split out the unit related data from the purely nonUnit listing data 

- [x] This change addresses the issue in full

## Description
On staging environment (the hosted one) hitting certain listing pages was causing the backend to run out of memory and crash. This is a fix to split out the unit related data from the purely nonUnit listing data 

## How Can This Be Tested/Reviewed?
On local nothing should really change, you should be able to view the list of listings, the listing itself, applications, and on the public side see the listing as well 

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by YazeedLoonat** [comment]
@ludtkemorgan got this updated

File: backend/core/src/listings/listings.service.ts [hunk 1]
```diff
@@ -150,7 +162,12 @@ export class ListingsService {
 
   async findOne(listingId: string, lang: Language = Language.en, view = "full") {
     const qb = getView(this.listingRepository.createQueryBuilder("listings"), view).getViewQb()
-    const result = await qb.where("listings.id = :id", { id: listingId }).getOne()
+    const fullListingDataQuery = qb.where("listings.id = :id", { id: listingId }).getOne()
+
+    const fullUnitDataQuery = this.getUnitsForListing(listingId)
+
+    const [result, unitData] = await Promise.all([fullListingDataQuery, fullUnitDataQuery])
+    result.units = unitData.units
```
Comment (ludtkemorgan) [inline backend/core/src/listings/listings.service.ts:171]: Would it make sense to make this a function since it's duplicated in 2 spots?

**Review by ludtkemorgan — APPROVED** [review]
Functionally this looks good, just one comment. I'm curious if you were able to capture any performance increases with this change?

**Comment by YazeedLoonat** [comment]
@ludtkemorgan 
Benchmarking on a local build is a little tricky since I have some other things running that can spike the mem usage. I ended up hitting  http://localhost:3100/listings?page=1&limit=100 to get these numbers. I think its pretty visible that the changes dropped the max mem usage, and antidotally it also returned faster 

control: 
image




with changes: 
image




without changes: 
image

File: backend/core/src/listings/listings.service.ts (hunk 1 above)
Comment (YazeedLoonat) [reply backend/core/src/listings/listings.service.ts]: nice call got this updated if you wanna peep

**Comment by YazeedLoonat** [comment]
Hey @ludtkemorgan got this updated per your notes

**Review by seanmalbert — APPROVED** [review]

**Pull request merged**